src/controllers/midi/midicontroller.cpp

```diff
@@ -275,23 +305,37 @@ void MidiController::processInputMapping(const MidiInputMapping& mapping,
             return;
         }
 
-        QJSValue function = pEngine->wrapFunctionCode(mapping.control.item, 5);
+        if (std::holds_alternative<ConfigKey>(mapping.control)) {
+            // Lazily compute the QJSValue
+            QJSValue function = pEngine->wrapFunctionCode(
+                    std::get<ConfigKey>(mapping.control).item, 5);
+            mapping.control = std::make_shared<QJSValue>(function);
+        }
+
+        VERIFY_OR_DEBUG_ASSERT(std::holds_alternative<std::shared_ptr<QJSValue>>(mapping.control)) {
+            return;
+        }
+
         const auto args = QJSValueList{
                 channel,
                 control,
                 value,
                 status,
-                mapping.control.group,
```

Comment:
Oh dear, I just noticed this: If I lazily compute the function, the `group` parameter is lost. This is a breaking change then.

Reply:
Also, [there's already a function cache](https://github.com/mixxxdj/mixxx/blob/main/src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp#L71) so is there really a need to lazily compute the function after all?